### PR TITLE
[RAC-6136] update RackHD from src code install docs in ubuntu 16.04

### PR DESCRIPTION
**Pull request by HQebupt**
Update RackHD from src code install docs in ubuntu 16.04.
@changev @PengTian0 @panpan0000

### CODE REVIEW

File: docs/rackhd/ubuntu_source_installation.rst [hunk 1]
```diff
@@ -1,6 +1,102 @@
...
+
+* eth1 for dhcp/pxe to boot/configure the machines
+
+edit the network:
+
+* eth0 - assign IP address as appropriate for the environment, or you can use DHCP
+
+* eth1 static ( 172.31.128.0/22 )
+
+  this is the 'default'. it can be changed, but more than one file needs to be changed.)
```
Comment (panpan0000) [inline docs/rackhd/ubuntu_source_installation.rst]: sorry
what do you mean by "this" ?
does it mean "eth1" or "172.31.128.0/22" ?

File: docs/rackhd/ubuntu_source_installation.rst [hunk 2]
```diff
@@ -1,6 +1,102 @@
...
+
+* ens192 for dhcp/pxe to boot/configure the machines
+
+edit the network:
+
+* ens160 - assign IP address as appropriate for the environment, or you can use DHCP
+
+* ens192 static ( 172.31.128.0/22 )
+
+.. code::
```
Comment (panpan0000) [inline docs/rackhd/ubuntu_source_installation.rst]: I believe below code is to be appended to ```/etc/network/interfaces```
but I didn't found it's written ?

File: docs/rackhd/ubuntu_source_installation.rst [hunk 3]
```diff
@@ -9,61 +105,154 @@ We will leverage the ansible roles created for the RackHD demonstration environm
...
     sudo apt-get update
     sudo apt-get dist-upgrade
-    sudo reboot
-
-    cd ~
-    git clone https://github.com/rackhd/rackhd
     sudo apt-get install ansible
-    cd ~/rackhd/packer/ansible
-    ansible-playbook -i "local," -K -c local rackhd_local.yml
+    sudo apt-get install -y g++ libkrb5-dev unzip
```
Comment (panpan0000) [inline docs/rackhd/ubuntu_source_installation.rst]: the reason why you give up ansible-playbook?

File: docs/rackhd/ubuntu_source_installation.rst [hunk 4]
```diff
@@ -9,61 +105,154 @@ We will leverage the ansible roles created for the RackHD demonstration environm
...
+          name: on-syslog
+          cwd:  ./on-syslog
+        - script: index.js
+          name: on-tftp
+          cwd:  ./on-tftp
+
+3. pm2 start
 
 .. code::
+    sudo pm2 start rackhd.yml
```
Comment (panpan0000) [inline docs/rackhd/ubuntu_source_installation.rst]: should cd ~/src first

**Comment by panpan0000** [comment]
can you give a summary about what you have changed comparing with original doc?

File: docs/rackhd/ubuntu_source_installation.rst (hunk 1 above)
Comment (HQebupt) [reply docs/rackhd/ubuntu_source_installation.rst]: “this” means all the eth0 & eth1.  It refers [6.2.2.1.1. NICs](http://rackhd.readthedocs.io/en/latest/rackhd/ubuntu_package_installation.html)

File: docs/rackhd/ubuntu_source_installation.rst (hunk 3 above)
Comment (HQebupt) [reply docs/rackhd/ubuntu_source_installation.rst]: The ansible-playbook doesn't work well. It occur some errors, such as 
```
TASK [mongodb : Install MongoDB] ***********************************************
failed: [local] => (item=[u'mongodb=1:2.6.10-1ubuntu2']) => {"cache_update_time": 0, "cache_updated": false, "failed": true, "item": ["mongodb=1:2.6.10-1ubuntu2"], "msg": "'/usr/bin/apt-get -y -o \"Dpkg::Options::=--force-confdef\" -o \"Dpkg::Options::=--force-confold\"   install 'mongodb=1:2.6.10-1ubuntu2'' failed: E: Version '1:2.6.10-1ubuntu2' for 'mongodb' was not found\n", "stderr": "E: Version '1:2.6.10-1ubuntu2' for 'mongodb' was not found\n", "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\n", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information..."]}

PLAY RECAP *********************************************************************
local                      : ok=5    changed=0    unreachable=0    failed=1
```
@changev

**Comment by HQebupt** [comment]
@changev  If the ansible playbook run with some errors, how can user solve the problems ? Do we tell user how to solve it ?

**Comment by HQebupt** [comment]
It is better to use ansible playbook.

**Pull request closed without merging**